js/lib/components/feedback/selection.tsx

```diff
@@ -0,0 +1,98 @@
+import { VirtualElement } from "@popperjs/core";
+import React, { useCallback, useEffect, useState } from "react";
+import Highlighter from "web-highlighter";
+
+import { HIGHLIGHT_STORAGE_KEY } from "../../entryPoints/feedback";
+import FeedbackModal from "./modal";
+import SelectionTooltip from "./tooltip";
+
+type SelectionRendererProps = { highlighter: Highlighter; stored: any[] | undefined };
+let SelectionRenderer: React.FC<SelectionRendererProps> = ({ highlighter, stored }) => {
+  // id of feedback highlight currently being hovered over
+  const [hovered, setHovered] = useState<false | string>(false);
+
+  // current highlighted range of text
+  const [currRange, setCurrRange] = useState<false | Range>(false);
+
+  // whether feedback modal is open
+  const [modalOpen, setModalOpen] = useState(false);
+
+  // update selected range when selection changes
+  const handleSelection = useCallback(() => {
```

Comment:
I would add a note explaining that `useCallback` is necessary to keep a stable reference to the callback so it works with `removeEventListener`